src/coinbase/asset.ts

```diff
@@ -1,61 +1,97 @@
+  public readonly networkId: string;
+  public readonly assetId: string;
+  public readonly contractAddress: string;
+  public readonly decimals: number;
+
   /**
-   * Converts an amount from the atomic value of the primary denomination of the provided Asset ID
-   * to whole units of the specified asset ID.
+   * Private constructor for the Asset class.
    *
-   * @param {Decimal} atomicAmount - The amount in atomic units.
-   * @param {string} assetId - The asset ID.
-   * @returns The amount in whole units of the asset with the specified ID.
+   * @param networkId - The network ID.
+   * @param assetId - The asset ID.
+   * @param contractAddress - The address ID.
+   * @param decimals - The number of decimals.
    */
-  static fromAtomicAmount(atomicAmount: Decimal, assetId: string): Decimal {
-    switch (assetId) {
-      case "eth":
-        return atomicAmount.div(WEI_PER_ETHER);
-      case "gwei":
-        return atomicAmount.div(WEI_PER_GWEI);
-      case "usdc":
-        return atomicAmount.div(ATOMIC_UNITS_PER_USDC);
-      case "weth":
-        return atomicAmount.div(WEI_PER_ETHER);
-      default:
-        return atomicAmount;
-    }
+  private constructor(
+    networkId: string,
+    assetId: string,
+    contractAddress: string,
+    decimals: number,
+  ) {
+    this.networkId = networkId;
+    this.assetId = assetId;
+    this.contractAddress = contractAddress;
+    this.decimals = decimals;
+  }
+
+  /**
+   * Returns the Asset ID.
+   *
+   * @returns The Asset ID.
+   */
+  getAssetId(): string {
+    return this.assetId;
   }
 
   /**
-   * Converts the amount of the Asset to the atomic units of the primary denomination of the Asset.
+   * Creates an Asset from an Asset Model.
    *
-   * @param amount - The amount to normalize.
-   * @param assetId - The ID of the Asset being transferred.
-   * @returns The normalized amount in atomic units.
+   * @param model - The Asset Model.
+   * @param assetId - The Asset ID.
+   * @throws If the Asset Model is invalid.
+   * @returns The Asset Class.
    */
-  static toAtomicAmount(amount: Decimal, assetId: string): Decimal {
-    switch (assetId) {
-      case "eth":
-        return amount.mul(WEI_PER_ETHER);
-      case "gwei":
-        return amount.mul(WEI_PER_GWEI);
-      case "usdc":
-        return amount.mul(ATOMIC_UNITS_PER_USDC);
-      case "weth":
-        return amount.mul(WEI_PER_ETHER);
-      default:
-        return amount;
+  public static fromModel(model: AssetModel, assetId?: string) {
+    if (!model) {
+      throw new InternalError("Invalid asset model");
     }
+
+    let decimals = model.decimals!;
+    // TODO: Push this logic down to the backend.
+    if (assetId && model.asset_id !== Coinbase.toAssetId(assetId)) {
+      switch (assetId) {
+        case "gwei":
+          decimals = GWEI_DECIMALS;
+          break;
+        case "wei":
+          decimals = 0;
+          break;
+        case "eth":
+          break;
+        default:
```

Comment:
default should be break. we are supporting more erc20s on mainnet now https://docs-cdp-developer-platform-preview.cbhq.net/wallets/docs/assets/

Reply:
this is only for ETH, Wei, or Gwei usage. ERC20 tokens should not be able to bypass the `model.asset_id !== Coinbase.toAssetId(assetId)` guard.